bazel_tools/client_server/tests/client.sh

```diff
@@ -27,6 +27,12 @@ if [[ -z "$port" || -z "$inputFile" ]]; then
   exit 1
 fi
 
+
 exec 5<> /dev/tcp/localhost/$port
 cat $inputFile >&5
 cat <&5
+
+# Write the output file for testing client_server_build.
+if [ -n "${CS_OUT-}" ]; then
```

Comment:
What's the significance of the `-` after the variable name?

Reply:
The script has `set -u` enabled, which means that `"${CS_OUT}"` would be an error if `CS_OUT` is not defined. With `"${CS_OUT-}"` (with the dash) the shell will substitute the empty string and not raise an error.

Reply:
thank you, @aherrmann-da